index/views.py

```diff
@@ -23,24 +23,27 @@
 '''
 import time
 import random
-import datetime
 from django.http import Http404
 from django.utils import timezone
 from utils.log_info import get_logger
 from contest.models import Contest
 from django.shortcuts import render
 from django.http import HttpResponse
+from datetime import datetime, timedelta
 from users.models import User, Notification
 from django.template import RequestContext
 from utils.user_info import validate_user
+from django.template import RequestContext
 
 # Create your views here.
 logger = get_logger()
 def index(request, alert_info='none'):
-
     present = timezone.now()
-    c_runnings = Contest.objects.filter(start_time__lt=present, end_time__gt=present)
-    c_upcomings = Contest.objects.filter(start_time__gt=present)
+    time_threshold = datetime.now() + timedelta(days=1);
+    c_runnings = Contest.objects.filter \
+        (start_time__lt=present, end_time__gt=present,end_time__lt=time_threshold)
```

Comment:
It makes no sense to ignore running contests which lasts more than one day.
BTW, I think we can ignore homework by add filter ```is_homework=false```.